harvest/trader/trader.py

```diff
@@ -406,6 +403,56 @@ def _update_stats(self, df_dict, new=False, option_update=False):
             equity = net_value + self.account['cash']
             self.account['equity'] = equity
 
+    def storage_setup(self, interval):
+        """Loads historical price data to ensure price storage are up-to-date upon program startup.
+        This should also be called at the start of each trading day 
+        so database is updated and cache is refreshed.
+        """
+        debug("Initializing storage...")
+        
+        start = pytz.utc.localize(dt.datetime(1970, 1, 1))
+        end = pytz.utc.localize(dt.datetime.utcnow().replace(microsecond=0, second=0))  # Current timestamp in UTC
+
+        for symbol in self.watch:
+            
+            df = self.streamer.fetch_price_history(start, end, interval, symbol)
+            self.storage.store(symbol, interval, df)
+
+            # Many brokers have separate API for intraday data, so make an API call
+            # instead of aggregating interday data
+            if '1DAY' in self.aggregations:
+                df = self.streamer.fetch_price_history(last, today, '1DAY', symbol)
+                self.storage.store(symbol, '1DAY', df)
+
+            df = self.storage.load(symbol, interval)
+            for i in self.aggregations:
+                if i == '1DAY':
+                    continue
+                df_tmp = self.aggregate_df(df, i)
+                self.storage.store(symbol, i, df_tmp)
+
+    def storage_update(self, df_dict):
```

Comment:
update the storage with new data

Reply:
Same for this, I took advantage of your Storage class and minimized the code in the Trader class: https://github.com/tfukaza/harvest/blob/462fcee387fa924709372545b2554a1795f913e6/harvest/trader/trader.py#L281-L288